File: 백준/Gold/2565. 전깃줄/전깃줄.py
N = int(input())

wire = []

for i in range(N):
    a, b = map(int, input().split(' '))

    wire.append([a,b])


wire.sort()


for i in range(N):
    wire[i] = wire[i][1]


def LIS(seq):

    dp = [1] * len(seq)
    dp[0] = 1

    for i in range(1, len(seq)):

        for j in range(0, i):

            if seq[j] < seq[i]:
                dp[i] = max(dp[i], dp[j]+1)

    return max(dp)

print(N - LIS(wire))

File: 백준/Gold/2565. 전깃줄/전깃줄.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int LIS(const vector<int>& seq) {
    int n = seq.size();
    vector<int> dp(n, 1);  // 모두 1로 초기화

    for (int i = 1; i < n; ++i) {
        for (int j = 0; j < i; ++j) {
            if (seq[j] < seq[i]) {
                dp[i] = max(dp[i], dp[j] + 1);
            }
        }
    }

    return *max_element(dp.begin(), dp.end());
}

int main() {
    int N;
    cin >> N;

    vector<pair<int, int>> wire(N);
    for (int i = 0; i < N; ++i) {
        cin >> wire[i].first >> wire[i].second;
    }

    // A 기준 정렬
    sort(wire.begin(), wire.end());

    // B만 추출
    vector<int> b_values(N);
    for (int i = 0; i < N; ++i) {
        b_values[i] = wire[i].second;
    }

    int result = N - LIS(b_values);
    cout << result << '\n';

    return 0;
}
